### ISSUE TITLE
Static ports for NFS/RPC

### ISSUE THREAD

**Issue by SunMar** [codekitchen/dinghy#260]
Hi,

Is it possible to configure the ports Dinghy listens on for NFS and RPC (and others too if there are more)? My firewall is going haywire constantly popping up that a new port was opened again (both the 19xxx for NFS and ports below 1024 which I think are RPC).

**Comment by codekitchen** [comment]
Is it possible to configure your firewall to always allow the `unfsd` binary? If so that would be the quickest solution. Otherwise:

There's not currently any way to configure the NFS port, but it wouldn't be too hard to add. Basically add a new preference in cli.rb and plumb it through to the [Unfs constructor](https://github.com/codekitchen/dinghy/blob/master/cli/dinghy/unfs.rb#L15). A better alternative might be to select the first port >= 19000 that isn't currently being used, rather than randomly selecting a port, but that's a bit more complex.

I'm not sure what the ports below 1024 would be, there's no other incoming connections. Or does your firewall prompt for outgoing connections as well? At what point do those connections occur? If it is an outgoing conn, it could be the fsevents daemon opening the SSH connection to the VM. Or potentially something docker-machine is doing.

**Comment by SunMar** [comment]
I've looked into it allowing a specific program but no luck so far.

From a security perspective I prefer to have control over which ports are open on my machine and not have an application pick one itself (whether that's random or looking for the first free port). You could perhaps still have "pick a random port" as the default, and only bind to a static port if one is configured. I don't really see the benefit of looking for the first available port, to me that's not much different from a random port and I don't really see the added benefit. For people who don't care about the port, selecting one at random is fine. People who do care about the port probably won't like it if despite configuration the application still binds to a different port.

Or am I missing something here? :)

As for the other ports below 1024, I'm starting to have doubts it's dinghy. I'm going to try to setup it up again from a clean MacOS somewhere in the next couple weeks to make sure and see if I can figure out more precise where the ports are coming from.

**Comment by codekitchen** [comment]
Hm it looks like I may be able to remove the weird random port selection now. I originally did that because I ran into issues with unfsd failing on `dinghy restart` trying to bind to the same port again, and I didn't want to dig into the unfsd source to try and fix it. But I'm not seeing that issue on restarts anymore, so it appears to have been fixed sometime in the last 2 years or so. I'm going to try that out for a while and see if anything crops up.
